### PR TITLE
4.2

### PR DESCRIPTION
**Pull request by artem328**
Updated config for 4.2

@etomilin, please, create branch "4.2", so I can merge changes to it

### CODE REVIEW

File: config/docker/php/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,43 @@
-FROM 4xxi/php-pgsql:flex
+FROM php:7.2-fpm-alpine
```
Comment (etomilin) [inline config/docker/php/Dockerfile]: I think we should start new projects from 7.3. 
Any arguments to use 7.2?

File: config/docker/php/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
+    postgresql-dev
+
+RUN docker-php-ext-install \
+    intl \
+    opcache \
+    pdo \
+    pdo_pgsql \
+    zip
+
+RUN EXPECTED_SIGNATURE="$(wget -q -O - https://composer.github.io/installer.sig)" \
```
Comment (eboboshka) [inline config/docker/php/Dockerfile]: Package installation can be replaced by copying a binary file from the official repository.
`COPY --from=composer /usr/bin/composer /usr/bin/composer`

File: .env.dist [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Docker configuration
+HOST_UID=
 SERVER_HTTP_PORT=80
+POSTGRES_EXTERNAL_PORT=5432
```
Comment (etomilin) [inline .env.dist]: I think that for consistency we should either use _EXTERNAL_ for both postgres and nginx or not to use it in both cases.

File: config/docker/php/Dockerfile [hunk 4]
```diff
@@ -1,11 +1,43 @@
...
-#    && docker-php-ext-install \
-#    ext_name_1 \
-#    ext_name_2
\ No newline at end of file
+RUN apk add --no-cache \
+    git \
+    zip \
+    icu-dev \
+    zlib-dev \
+    postgresql-dev
```
Comment (etomilin) [inline config/docker/php/Dockerfile]: packages should be sorted by name (to prevent duplicates in long lists).

File: docker-compose.yml [hunk 5]
```diff
@@ -5,11 +5,17 @@ services:
         build: ./config/docker/php
         depends_on:
             - db
+        working_dir: /var/www/html
```
Comment (etomilin) [inline docker-compose.yml:11]: We can move this inside dockerfile

File: docker-compose.yml [hunk 6]
```diff
@@ -5,11 +5,17 @@ services:
...
         volumes:
             - .:/var/www/html
         expose:
             - "9000"
-        entrypoint: ./config/docker/docker-entrypoint.sh
+        entrypoint: ./config/docker/php/docker-entrypoint.sh
```
Comment (etomilin) [inline docker-compose.yml:16]: This could be also moved into dockerfile

File: config/docker/php/Dockerfile [hunk 7]
```diff
@@ -1,11 +1,43 @@
-FROM 4xxi/php-pgsql:flex
+FROM php:7.2-fpm-alpine
 
-# To add additional packages and php extensions uncomment the following lines and add required dependencies
-# Check https://hub.docker.com/_/php/ for examples and documentation
+COPY ./symfony.ini /usr/local/etc/php/conf.d/
+COPY ./symfony.pool.conf /usr/local/etc/php-fpm.d/
```
Comment (ipopov4xxi) [inline config/docker/php/Dockerfile]: If symfony.ini of symfony.pool.conf will be changed whole image must be rebuilded. Better make this operations at the end of Dockerfile, so rebuild will affect only latest layers.

File: docker-compose.yml [hunk 8]
```diff
@@ -5,11 +5,17 @@ services:
...
+        environment:
+            - COMPOSER_HOME=/var/www/html/var/composer
```
Comment (etomilin) [inline docker-compose.yml:18]: Why do we need this?

File: config/docker/php/docker-entrypoint.sh [hunk 9]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+COMMAND="composer install --prefer-dist --no-progress --no-suggest --no-interaction"
+
+if [[ "prod" = "${APP_ENV}" ]]; then
+    COMMAND="${COMMAND} --no-dev  --optimize-autoloader --classmap-authoritative"
+fi
+
+if [[ ! -z "${HOST_UID}" ]]; then
+    deluser www-data
```
Comment (ipopov4xxi) [inline config/docker/php/docker-entrypoint.sh]: No need to delete user. You can modify by it `/usr/sbin/usermod -u "$HOST_UID" www-data`

File: config/docker/php/Dockerfile [hunk 10]
```diff
@@ -1,11 +1,43 @@
...
+        >&2 echo 'ERROR: Invalid installer signature' \
+        && rm composer-setup.php \
+        && exit 1; \
+    fi; \
+    php composer-setup.php --install-dir=/usr/local/bin --filename=composer; \
+    RESULT=$? \
+    && rm composer-setup.php \
+    && exit ${RESULT}
+
+# Uncomment code below for enabling xDebug
```
Comment (aburov-dev) [inline config/docker/php/Dockerfile:17]: Еще не плохо бы добавить сюда добавление GD для работы с картинками, можно даже закомменченный код.

File: config/docker/php/docker-entrypoint.sh [hunk 11]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+COMMAND="composer install --prefer-dist --no-progress --no-suggest --no-interaction"
+
+if [[ "prod" = "${APP_ENV}" ]]; then
+    COMMAND="${COMMAND} --no-dev  --optimize-autoloader --classmap-authoritative"
+fi
+
+if [[ ! -z "${HOST_UID}" ]]; then
```
Comment (aburov-dev) [inline config/docker/php/docker-entrypoint.sh]: А это можно решить на уровне Dockerfile а не entrypoint

Comment (ipopov4xxi) [reply config/docker/php/docker-entrypoint.sh]: Это нельзя решить на уровне Dockerfile. В Dockerfile не передать env переменные.

Comment (rk4xxicom) [reply config/docker/php/docker-entrypoint.sh]: можно через build-args

File: docker-compose.yml [hunk 12]
```diff
@@ -20,9 +26,9 @@ services:
             - .:/var/www/html
             - ./var/log:/var/log/nginx
     db:
-        image: postgres:9
+        image: postgres:9.6
```
Comment (etomilin) [inline docker-compose.yml]: I think we can upgrade to 10.x 
@atrutko-4xxi any concerns from your side?

Comment (eboboshka) [reply docker-compose.yml]: @artem328 I agree with @etomilin , you need to start using version 10+
The main limitation on using Postgres version 10 is doctrine/dbal v2.7.0+
https://github.com/doctrine/dbal/releases

**Comment by eboboshka** [comment]
@artem328 Don't forget about it https://docs.docker.com/engine/reference/builder/#dockerignore-file ;-)

File: config/docker/php/Dockerfile (hunk 2 above)
Comment (ipopov4xxi) [reply config/docker/php/Dockerfile]: Maybe also add https://github.com/hirak/prestissimo to speedup composer install by default?

Comment (dyudin4xxi) [reply config/docker/php/Dockerfile]: Yes, I agree
This will help speed up the installation of dependencies and container assembly.

**Comment by artem328** [comment]
@atrutko-4xxi does it make sense on development environment?

**Comment by eboboshka** [comment]
> @atrutko-4xxi does it make sense on development environment?

Of course.
For example, why the .git directory is needed in the Docker image?
Second, it will force developers to use the recommendations of the Docker community.

**Pull request merged**